### PR TITLE
Add optional `params` key to yup `createError()` argument

### PR DESCRIPTION
**Pull request by ericbiewener**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jquense/yup/pull/709

### CODE REVIEW

**Review by sseppola — COMMENTED** [review]
Hello and thank you for your PR!
This looks good to me, only thing I'm missing is a test, is this something you could add to this PR?

I'm happy for this to be merged, however this helps us with future contributions :)

**Comment by YashdalfTheGray** [comment]
@ssepola can you mark your comment as request changes? I think we should block on tests being there for sure.

**Review by YashdalfTheGray — CHANGES_REQUESTED** [review]
@sseppola requested that we add tests for this. I agree too.

**Comment by ericbiewener** [comment]
Done! Let me know if what I added is sufficient for the tests.

File: types/yup/yup-tests.ts [hunk 1]
```diff
@@ -214,7 +214,7 @@ yup.mixed<string>().concat(yup.date()); // $ExpectType MixedSchema<string | Date
 
 // Async ValidationError
 const asyncValidationErrorTest = function(this: TestContext): Promise<ValidationError> {
-    return new Promise(resolve => resolve(this.createError({ path: 'testPath', message: 'testMessage' })));
+    return new Promise(resolve => resolve(this.createError({ path: 'testPath', message: 'testMessage', params: { foo: 'bar' } })));
```
Comment (YashdalfTheGray) [inline types/yup/yup-tests.ts]: Since it's optional, can we add a test case for when it's not present also?

**Review by YashdalfTheGray — CHANGES_REQUESTED** [review]

**Comment by YashdalfTheGray** [comment]
@ericbiewener I just have one small change that I left in the comments!

**Comment by ericbiewener** [comment]
All the options for `createError` are optional, so I added tests the excludes all options, not just the new `params` key.

**Comment by ericbiewener** [comment]
All the options for `createError` are optional, so I added tests the excludes all options, not just the new `params` key.

**Review by YashdalfTheGray — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**